example/plugin/main.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+/*
+ * SPDX-FileCopyrightText: 2023 Siemens AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Author: Michael Adler <michael.adler@siemens.com>
+ */
+
+import (
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/siemens/wfx/generated/plugin"
+	"github.com/siemens/wfx/generated/plugin/client"
+	"github.com/siemens/wfx/middleware/plugin/ioutil"
+)
+
+const queueSize = 64
+
+type QueueEntry struct {
+	start   time.Time
+	request *plugin.PluginRequestT
+}
+
+func main() {
+	// NOTE: The working directory for this process is inherited from the wfx parent process
+	file, err := os.OpenFile("plugin.log", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o644)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file.Close()
+	log.SetOutput(file)
+	log.Println("[INFO] Plugin starting...")
+
+	queue := make(chan QueueEntry, queueSize)
+
+	// reader goroutine: it's only purpose is to read from stdin as soon as
+	// something comes in, to prevent the other side from blocking
+	go func() {
+		for {
+			req, err := ioutil.ReadRequest(os.Stdin)
+			if err != nil {
+				log.Println("[ERROR] Failed to receive message", err)
+				continue
+			}
+
+			entry := newQueueEntry(req)
+			// this is important: we rather discard the message instead of
+			// blocking, since we want to read the next message from stdin
+			select {
+			case queue <- entry:
+				log.Println("[DEBUG] Message enqueued")
+				// success
+			default:
+				log.Println("[ERROR] Queue full. Message discarded.")
```

Comment:
In this case the message is lost and wfx doesn't get an answer to this `cookie`. What happens then? Shouldn't a plugin behave benign and send something it deems appropriate, e.g., deny?

Reply:
Yep good point